### PR TITLE
Add support for Cassandra Vector Store

### PR DESCRIPTION
**Pull request by alexleventer**
- Add support for Cassandra vector store to Langchain.js
- Test coverage for Cassandra vector store

Run integration tests with `yarn test:single langchain/src/vectorstores/tests/cassandra.int.test.ts`

### CODE REVIEW

**Comment by jacoblee93** [comment]
Thank you @alexleventer!

I can add some docs for this - looks like we can set something up with AstraDB?

https://astra.datastax.com/org/272028b1-ae74-44c9-961c-25ed054a8764/database/5083e144-e541-4098-a678-638e88a1a7d7/connect

**Comment by jacoblee93** [comment]
Having a bit of trouble actually - keep running into this from a fresh AstraDB instance:

```
  ● CassandraStore › CassandraStore.fromText

    ResponseError: Missing correct permission on test.test

      at FrameReader.readError (../node_modules/cassandra-driver/lib/readers.js:389:17)
      at Parser.Object.<anonymous>.Parser.parseBody (../node_modules/cassandra-driver/lib/streams.js:209:66)
      at Parser.Object.<anonymous>.Parser._transform (../node_modules/cassandra-driver/lib/streams.js:152:10)
      at Protocol.Object.<anonymous>.Protocol.readItems (../node_modules/cassandra-driver/lib/streams.js:124:10)
      at Protocol.Object.<anonymous>.Protocol._transform (../node_modules/cassandra-driver/lib/streams.js:47:10)
```

**Comment by alexleventer** [comment]
Yes - exactly @jacoblee93. You'll need to create an astra vector database to test. You can do so for free here: astra.datastax.com/register

**Comment by jacoblee93** [comment]
Yes, I am getting the above error when trying to run the integration tests and can't debug.

Are there other instructions I'm missing?

**Comment by alexleventer** [comment]
Hey @jacoblee93 - to test you'll need to first create an astra vector database and then set CASSANDRA_SCB and CASSANDRA_TOKEN env vars in your .env

**Comment by alexleventer** [comment]
Any luck @jacoblee93 ?

**Comment by jacoblee93** [comment]
Hey @alexleventer, I wasn't able to figure out how to use this as I kept running into the above permission error - would you be willing to write some docs that I could follow to set this up from scratch?

**Comment by alexleventer** [comment]
Yeah sure - would you mind sharing a link to the where you'd like the docs to live?

**Comment by jacoblee93** [comment]
This would be the best place right now:

https://github.com/langchain-ai/langchainjs/tree/main/docs/extras/modules/data_connection/vectorstores/integrations

We're planning to re-merge the skeleton stuff to make it clearer in the future.

**Comment by alexleventer** [comment]
Docs added @jacoblee93

**Review by Jack-Byrne — APPROVED** [review]

File: docs/extras/modules/data_connection/vectorstores/cassandra.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+5. Install the Cassandra Node.js driver.
+```bash npm2yarn
+npm install cassandra-driver
+```
+
+## Index and query docs
+
+```typescript
+import { Cassandra } from "langchain/vectorstores/milvus";
```
Comment (jacoblee93) [inline docs/extras/modules/data_connection/vectorstores/cassandra.md:36]: Change to `cassandra`

**Pull request merged**